File: typescript/basics/fizzbuzz.ts
// Print FizzBuzz for 1 through 100, with the label picked by a typed helper function.
// Run: tsc --target es2020 fizzbuzz.ts && node fizzbuzz.js

function fizzbuzzLabel(n: number): string {
    if (n % 15 === 0) return "FizzBuzz";
    if (n % 3 === 0) return "Fizz";
    if (n % 5 === 0) return "Buzz";
    return String(n);
}

for (let n = 1; n <= 100; n++) {
    console.log(fizzbuzzLabel(n));
}

File: rust/basics/fizzbuzz.rs
// Print FizzBuzz for 1 through 100, branching with match on a tuple of remainders.
// Build: rustc -o fizzbuzz fizzbuzz.rs

fn main() {
    for n in 1..=100 {
        let line = match (n % 3, n % 5) {
            (0, 0) => "FizzBuzz".to_string(),
            (0, _) => "Fizz".to_string(),
            (_, 0) => "Buzz".to_string(),
            _ => n.to_string(),
        };
        println!("{}", line);
    }
}
